security/mobile-app-network-security.md

```diff
@@ -0,0 +1,126 @@
+# Abstract
+
+In this document we will discuss some of the common problems around network security for mobile developers consuming AeroGear services in their apps, and propose a solution that will help resolve those problems.
+
+We will first look at some of the use cases in order to define the problems, and then look at some of the options to solve them. Finally the proposed solution will be presented.
+
+# Goals
+
+## Primary Goal
+
+* Reach agreement on a solution that can help mobile developers properly handle network security in their mobile apps that consume AeroGear services
+
+## Secondary Goal
+
+* Provide use cases and try to explain the problems to audiences that may not be familiar with the area, to help them understand and maybe decide the business value of the solution
+
+# Terms
+
+* CA: Certificate Authority. An entity that can issue digital certificates. They are trusted third-parties.
+* Self-signed Certificate: A self-signed certificate is an identity certificate that is signed by the same entity whose identity it certifies.
+* Man-in-the-middle attack: It is an attack where the attacker secretly relays and possibly alters the communication between two parties who believe they are directly communicating with each other.
+
+# Use Cases
+
+The main use cases that will be discussed in this document are:
+
+1. As a mobile developer, I want to use AeroGear services in my mobile apps. I have the AeroGear service running locally with HTTPS, and I want to test against the local services from my mobile app on an actual device.
+
+2. As an enterprise mobile developer, I need to develop a mobile app for internal use. I need to make sure my mobile app can connect to AeroGear services that are running with the organisation's own HTTPS certificates.
```

Comment:
Should we use the terms from above and say 'organisation's Self-signed certificate' instead of 'organisation's own HTTPS certificates'

Reply:
sure